src/nunjucksHelpers/mixin.ts

```diff
@@ -1,22 +1,42 @@
 import upath from 'upath';
 import fs from 'fs-extra';
 import nunjucks from 'nunjucks';
+import { pathInjector } from '../pathInjector';
 
 const mixinDirectoryKey = 'mixinDirectory';
 
+function parseMixinPath(mixinPath: string, pathInjector: pathInjector, currentFilePointer: string): string {
+  // Don't forget to parse the mixin path, as that could have been defined as an absolute path
+  // Maybe should be done when setting mixinPath initially
+  // eslint-disable-next-line prefer-const
+  let [parsedMixinPath, absolute] = pathInjector.injectMixin(mixinPath);
+
+  if (absolute) {
+    parsedMixinPath = upath.relative(upath.dirname(currentFilePointer), parsedMixinPath);
+  }
+
+  return upath.dirname(upath.normalize(parsedMixinPath));
+}
+
 export default function (): string {
   const tplGlobals = this.env.globals;
+  const pathInjector = (tplGlobals.pathInjector) as pathInjector;
+
   // eslint-disable-next-line prefer-rest-params
-  const renderPath = upath.join(upath.dirname(tplGlobals.currentFilePointer), arguments[0]);
+  const [argumentPath, isAbsolute] = pathInjector.injectMixin(arguments[0])
+
+  const renderPath = isAbsolute
```

Comment:
This logic might be better inside the pathInjector. I haven't checked if it's consistently applied everywhere injectMixin is used.

Reply:
This logic isn't the same in the two cases it's used - so would confuse things to move it into the function.